### DISCUSSION TITLE
Why the socket.IO example uses socket on the _app.js?

### DISCUSSION THREAD

**Discussion by theprobugmaker** [vercel/next.js#11090] [Help] [answered]
I installed the socket.IO example and I noticed that the socket instance is made in two places, on the hook and also on the `_app.js` the thing is that whenever I import the hook it will make a connection automatically and it will also make a connection because of `_app.js` (the server receives 2 connections), what is the reason to use in both places?

**Comment by theprobugmaker** [comment]
cc @timneutkens @chibicode

**Comment by theprobugmaker** [comment]
Anyone knows this?

**Comment by timneutkens** [comment, marked as answer]
The example was removed as it shows outdated practices and was generally not ideal

You can read more here https://twitter.com/rauchg/status/1237861882477723650

**Reply by theprobugmaker** [reply]
Thank you for the reply @timneutkens. I guess I'll have to go with another alternative then.
